packages/backend/src/workers/provider/LlamaCppPython.ts

```diff
@@ -173,7 +173,13 @@ export class LlamaCppPython extends InferenceProvider {
 
         // label the container
         labels['gpu'] = gpu.model;
-        envs.push(`GPU_LAYERS=${config.gpuLayers}`);
+
+        // set the number of layers to offload to the GPU in case of LibKrun
+        if (vmType === VMType.LIBKRUN || vmType === VMType.LIBKRUN_LABEL) {
+          envs.push(`GPU_LAYERS=999`);
```

Comment:
question (non-blocking): `-1` does not work as max?

Reply:
no unfortunately it doesn't

at the beginning I had errors as the 'cpu part' was used and not the GPU

comparing the arguments to start llama-server and AI Lab lead me to this difference


in RamaLama docs there is:

**ngl**=-1
number of gpu layers, 0 means CPU inferencing, 999 means use max layers (default: -1)
The default -1, means use whatever is automatically deemed appropriate (0 or 999)

(so it seems it's picking up 0 :-/ )

Reply:
and https://github.com/containers/ramalama/pull/802